chapter8.adoc

```diff
@@ -1,37 +1,124 @@
-is context switched (along with rest of the `msdcfg`) per SD.
+This chapter describes two extensions `Smsdedbg` and `Smsdetrc` that enable a
+RDSM to manage isolation of supervisor domains when external debug and trace
+mechanisms are being utilized. `Smsdedbg` defines the `sdedbgalw` bit in the
+<<MSDCFG>> CSR that specifies if a supervisor domain is allowed to be
+externally-debugged. Similarly, `Smsdetrc` defines the `sdetrcalw` bit in the
+<<MSDCFG>> CSR that specifies if a supervisor domain is allowed to be
+externally-traced cite:[ExtDbg]. External trace refers to tracing software
+running on a separate target RISC-V platform, via a trace control/data transport
+that provides external access to the trace encoder controls cite:[ETrc]. These
+two extensions only affect debug and trace orchestrated by an external actor -
+self-hosted debug (and trace) are managed directly by the RDSM. This chapter
+refers to the Debug specification cite:[ExtDbg], Trace specification
+cite:[ETrc], and the External debug security extension cite:[ExtDbgSec] for the
+description and behavior of controls outside the scope of this specification,
+but which interact with controls specified in this specification when one or
+more supervisor domains are used. Note that `Smsdedbg` and `Smsdetrc` may be
+used as independent extensions even when a single supervisor domain is being
+utilized on a platform.
 
-=== `Smsdedbg` interaction with external debug security controls (Informative)
+[NOTE]
+====
+The configuration for `msdcfg.sdedbgalw` and `msdcfg.sdetrcalw` is expected to
+be obtained from the manifest/configuration of the supervisor domain and
+reported in the attestation report for the supervisor domain and it's workloads.
+The RDSM should manage this supervisor domain configuration state for context
+switching using secure memory.
+====
+
+=== `Smsdedbg`: External Debug allowed control for Supervisor Domain
+
+When M-mode external debug is enabled, all supervisor domains may also be
+debugged by an external debugger irrespective of the configuration held in
+`msdcfg.sdedbgalw`.
+
+When M-mode external debug is disabled, whether execution at privilege modes
+less than `M-mode` may be debugged by an external debugger depends on the
+configuration held in `msdcfg.sdedbgalw`, as described below:
+
+When `msdcfg.sdedbgalw` = 0, external debug is disallowed for the supervisor
+domain. Abstract commands and halt request from the debug module are suppressed
+and stay pending while the supervisor domain is active, on a per-hart basis.
+Other debug operations (current or defined in the future) must similarly be kept
+pending while the supervisor domain is active on a hart. Triggers for the
+supervisor domain must be controlled by the RDSM to prevent any leakage of
+information across security domains.
+
+When `msdcfg.sdedbgalw` = 1 then external debug of privilege modes less than
+`M-mode` is allowed for such a supervisor domain on a per-hart basis. +
+
```

Comment:
When msdcfg.sdedbgalw is 1:
1. Access by external debuggers to the memory and/or state of the supervisor domain is allowed.
2. Entry to Debug Mode from a supervisor domain is allowed.

Reply:
added the 2 points in the requirements below.